\documentclass[11pt,english,letterpaper,oneside]{article}
\usepackage{amsmath,amssymb,amsfonts,amsthm,enumerate,enumitem}
\usepackage[top=3cm,bottom=3cm,left=2.5cm,right=2.5cm]{geometry}
\usepackage{graphicx,microtype}
\usepackage[T1]{fontenc}
\DisableLigatures[f]{encoding=T1}
\usepackage[skip=2pt]{caption}
\usepackage{hyperref,lineno}
\linenumbers
\usepackage{textcomp}
\usepackage{setspace,array,float}
\usepackage{bm,upgreek}
\usepackage[compact]{titlesec}
\usepackage[none]{hyphenat}
\setlength{\parskip}{0cm}
\usepackage{natbib}
\setcitestyle{citesep={;},aysep={}}
\usepackage{babel}
\usepackage{datetime}
\pagestyle{plain}
\frenchspacing
%\doublespacing
\usepackage{sectsty}
\usepackage[table]{xcolor}
\allsectionsfont{\bfseries}

\makeatletter
\g@addto@macro\normalsize{%
  \setlength\belowdisplayskip{15pt}
  \setlength\belowdisplayshortskip{10pt}
}
\makeatother

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\newcommand{\tmat}{$\bm{T}$}
\newcommand{\rmat}{$\bm{R}$}
\newcommand{\etal}{\textit{et al}.}
\newdateformat{mydate}{\THEDAY{} \monthname[\THEMONTH], \THEYEAR}

\hfill Version dated: \mydate\today
\vspace{0.25in}

\begin{center}

{\LARGE \bfseries Estimating allele frequencies in non-model polyploids using high throughput sequencing data}
\vspace{0.45in}

Paul D. Blischak$^{1,\dagger}$, Laura S. Kubatko$^{1,2}$ and Andrea D. Wolfe$^1$
\vspace{0.45in}


\textit{$^1$Department of Evolution, Ecology and Organismal Biology, The Ohio State University,}

\textit{318 W. 12th Avenue, Columbus, OH 43210, USA.}
\bigskip
\bigskip

\textit{$^2$Department of Statistics, The Ohio State University,}

\textit{1958 Neil Avenue, Columbus, OH 43210, USA.}


\end{center}
\vspace{0.45in}


\noindent $^\dagger$\textbf{Corresponding author}: Paul Blischak, Dept. of Evolution, Ecology and Organismal Biology, Aronoff Laboratory, The Ohio State University, 318 W. 12th Avenue, Columbus, OH 43210. E-mail: blischak.4@osu.edu.

\vspace{0.45in}

\noindent \textbf{Running title}: Allele frequencies in polyploids

\vspace{.45in}

%%%%%%%%%%%%%%%%%%%%
\section*{Abstract}                      %%
%%%%%%%%%%%%%%%%%%%%

Despite the ever increasing opportunity to collect large-scale datasets for population genomic analyses, the use of high throughput sequencing to study populations of polyploids has seen little application. This is due in large part to problems associated with determining allele copy number in the genotypes of polyploid individuals (allelic dosage uncertainty--ADU), which complicates the calculation of important quantities such as allele frequencies. This well known problem has hindered population genetic studies in polyploids for decades, though several tools exist for analyzing genetic data from polyploids by dealing with particular issues of ADU. Additional complications arise because of the mixed inheritance patterns and variable reproductive modes that are characteristic of many polyploid taxa, making the development of population genetic models for polyploids especially difficult. Here we describe a statistical model to estimate biallelic SNP frequencies in a population of polyploids using high throughput sequencing data in the form of read counts. Uncertainty in the number of copies of an allele in an individual's genotype is accounted for by treating genotypes as an intermediate, latent variable in a hierarchical Bayesian model. In this way, we bridge the gap from data collection (using techniques such as restriction-site associated DNA sequencing) to allele frequency estimation in a unified inferential framework by summing over genotype uncertainty. Simulated datasets were generated under various conditions for both tetraploid and hexaploid populations to evaluate the model's performance and to help guide the collection of empirical data. We also discuss potential sources of bias that could influence results, as well as propose model extensions to ameliorate some of these biases.
\vspace{0.25in}

\noindent (\textbf{Keywords}: allelic dosage uncertainty, allele frequencies, hierarchical Bayesian modeling, polyploidy, population genomics)
\vspace{0.25in}

%%%%%%%%%%%%%%%%%%%
\section*{Introduction}            %%
%%%%%%%%%%%%%%%%%%%

Biologists have long been fascinated by the occurrence of whole genome duplication (WGD) in natural populations and have recognized its role in the generation of biodiversity \citep{ClausKeckHies1940,StebbinsVariationEvolution,GrantPlantSpeciation,otto2000polyploidy}. Though WGD is thought to have occurred at some point in nearly every branch of the Tree of Life (plants, animals and fungi), it is a particularly common phenomenon in plants and is regarded by many to be an important factor in plant diversification \citep{wood2009polyploid,soltisd2009diversification,scarpino2014polyploid}. The role of polyploidy in plant evolution was originally considered by some to be a ``dead-end'' \citep{StebbinsVariationEvolution,wagner1970noise,soltisd2014stebbins} but, since its first discovery in the early twentieth century, polyploidy has been continually studied in nearly all areas of botany \citep{winge1917polyploidy,Winkler1916polyploidy,ClausKeckHies1945polyploidy,GrantPlantSpeciation,StebbinsVariationEvolution,soltisD2003polyploid,soltisd2010polyploidUnknowns,soltai2009roleOfHybridization,ramsey2014polEcoProcRoySoc}. Though fewer examples of WGD are currently known for animal systems, groups such as amphibians, fish, and reptiles all exhibit polyploidy \citep{allendorf1984tetraploidFish,gregory2005polyploidyAnimals}. Ancient genome duplications are also thought to have played an important role in the evolution of both plants and animals, occurring in the lineages preceeding the seed plants, angiosperms and vertebrates \citep{ohno1970geneDuplication,otto2000polyploidy,furlong2001animalOctoploid,jiao2011ancientWGD}. These ancient WGD events during the early history of seed plants and angiosperms have also been followed by several more WGDs in all major plant groups \citep{cui2006genomeDuplication,scarpino2014polyploid,canon2014polyploidyLegumes}. Recent experimental evidence has also demonstrated increased survivorship and adaptability to foreign environments of polyploid taxa when compared with their lower ploidy relatives \citep{ramsey2011polyploidEcology,Selmecki2015yeastAdaptation}.
\medskip

Polyploids are generally broken into two types based on how they are formed: auto- and allopolyploids. Autopolyploids form when a WGD event occurs within a single evolutionary lineage. Allopolyploids are formed by hybridization between two separately evolving lineages followed by WGD \citep{StebbinsVariationEvolution}. 
\medskip

The theoretical treatment of population genetic models in polyploids has it origins in the Modern Synthesis with  Fisher, Haldane and Wright each contributing to the development of some of the earliest mathematical models for understanding the genetic patterns of inheritance in polyploids. Among the earliest of these works was Haldane's 1930 paper on autopolyploid inheritance in $2k$-ploid ($k=2,3,\dots$) organisms. Influenced in part by the works of Hermann J. Muller in tetraploid species of \textit{Primula} (\citeyear{muller1914primula}) and W. J. C. Lawrence in octoploid species of \textit{Dahlia} (\citeyear{lawrence1929dahlia}), Haldane generalized the combinatorial formulas for determining the frequencies of the different possible gametes formed from all genotype combinations for a $2k$-ploid. He also considered additional factors influencing gamete frequencies such as double reduction and the effects of partial selfing \citep{haldane1930autopolyploids}. Fisher's interest in polyploidy stemmed largely from observations made in the plant genus \textit{Lythrum}, which exhibited conspicuous patterns of trimorphic heterostyly \citep{fisher1941lythrum}. Empirical works by Nora Barlow (\citeyear{barlow1913heterostylism}, \citeyear{barlow1923trimorphic}), as well as initial investigations into the inheritance patterns of the three style types (Short, Mid, Long) by E. M. East (\citeyear{east1927lythrum}) formed the basis for Fisher's formulation of a model for the inheritance patterns of the Mid length style form in \textit{Lythrum salicaria} \citep{fisher1941lythrum}. He later added to this work by considering double reduction in the inheritance of the Mid length style and complemented his theoretical work through a collaboration with Kenneth Mather to complete crossing experiments \citep{fisher1943doublereduction,fisher1943mather}. Wright's contributions were concerned with the calculation of the distribution of allele frequencies in a $2k$-ploid and were largely an extension of his classic 1931 paper, \textit{Evolution in Mendelian populations}, and a previously published manuscript describing similar processes in diploids \citep{wright1931mendelianPops,wright1937geneFreqs,wright1938polyploid}. Wright was among the first to consider mutation, migration, selection and inbreeding in his formulation of the distribution of gene frequencies, which helped to establish future ideas about modeling allelic diffusion in a population. For example, it was noted by Motoo Kimura that much of the work on diffusion equations in population genetics could be applied to polyploids in a manner similar to Wright's derivation of the allele frequency distribution in polyploids \citep{kimura1964diffusion}.
\medskip

The foundation laid down by these early papers has led to the continuing development of population genetic models for polyploids, including models for understanding the rate of loss of genetic diversity and extensions of the coalescent in autotetraploids, as well as modifications of the multispecies coalescent for the inference of species networks containing allotetraploids \citep{moody1993autopolyploids,arnold2012autotetraploidCoal,jones2013allopolyploid}. Much of this progress was described in a review by Dufresne \etal{} (\citeyear{dufresne2014polyPopGen}), who outlined the current state of population genetics in polyploids regarding both molecular techniques and statistical models. Not surprisingly, one of the most promising developments for the future of population genetics in polyploids is the advancement of sequencing technologies. A particularly popular method of gathering large datasets for genome scale inferences is restriction-site associated DNA sequencing [RADseq] \citep{miller2007gbs,baird2008radTags,puritz2014demystifyingRAD}. However, despite its popularity for population genetic inferences at the diploid level, there are much fewer examples of RADseq experiments conducted on polyploid taxa \citep[but see][]{ogden2013sturgeonRADseq,wang2013birchRADseq,logan-young2015polyploidSNP}. Among the primary reasons for the dearth in applying RADseq to polyploids is the issue of allelic dosage uncertainty (ADU), or the inability to fully determine the genotype of a polyploid organism when it is partially heterozygous. For a biallelic locus (allele A or B), a partially heterozygous polyploid will have high throughput sequencing reads containing both the A and the B allele. For a tetraploid, the possible genotypes are AAAB, AABB and ABBB. For a hexaploid they are AAAAAB, AAAABB, AAABBB, AABBBB and ABBBBB. In general, the number of possible genotypes for a biallelic locus of a partially heterozygous N-ploid ($\text{N}=3,4,5,\ldots$) is $\text{N}-1$. A possible solution to this problem would be to develop . However, this could lead to erroneous inferences when genotypes are simply fixed at point estimates based on read proportions without considering estimation error. Furthermore, when sequencing coverage is low, the number of genotypes that will appear to be equally probable increases with ploidy, making it difficult to distinguish among the possible partially heterozygous genotypes.
\medskip

In this paper we describe a model that aims to address the problems associated with ADU by treating genotypes as a latent variable in a hierarchical Bayesian model and using read counts as data. In this way, we preserve the uncertainty that is inherent in the genotypes of partially heterozygous polyploids by inferring a probability distribution across all possible values of the genotype, rather than treating the genotypes as being directly observed. Our model assumes that the ploidy level of the population is known and that the genotypes of individuals in the population are drawn from a single underlying allele frequency for each locus. These assumptions imply that alleles in the population are undergoing polysomic inheritance without double reduction, which most closely adheres to the inheritance patterns of an autopolyploid. We acknowledge that the model in its current form is an oversimplification of biological reality and realize that it does not apply to a large portion of polyploid taxa. Nevertheless, we believe that accounting for ADU by modeling genotype uncertainty has the potential to be applied more broadly via modifications of the probability model used for the inheritance of alleles, which could lead to more generalized population genetic models for polyploids (see the \textbf{Extensibility} section of the \textbf{Discussion}). 
\medskip

%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Materials and Methods}        %%
%%%%%%%%%%%%%%%%%%%%%%%%

\noindent Our goal is to estimate the frequency of a reference allele for each locus sampled from a population of known ploidy ($\psi$), where the reference allele can be chosen arbitrarily between the two alleles at a given biallelic SNP. To do this we extend the population genomic models of \cite{buerkle2013popModels}, which employ a Bayesian framework to model high throughput sequencing reads ($\bm{T},\bm{R}$), genotypes ($\bm{G}$) and allele frequencies ($\bm{p}$), to the case of arbitrary ploidy. The idea behind the model is to view the sequencing reads gathered for an individual as a random sample from the unobserved genotype at each locus. Genotypes can then be treated as a parameter in a probability model that governs how likely it is that we see a particular number of sequencing reads carrying the reference allele. Similarly, we can treat genotypes as a random sample from the underlying allele frequency in the population (assuming Hardy-Weinberg equilibrium). This hierarchical setup addresses the problems associated with ADU by treating genotypes as a latent variable that can be integrated out using Markov chain Monte Carlo (MCMC).

\medskip
\subsection*{Model setup}
\medskip

Here we consider a sample of $N$ individuals from a single population of ploidy level $\psi$ sequenced at $L$ unlinked SNPs. The data for the model consist of two matrices containing counts of high throughput sequencing reads mapping to each locus for each individual: \rmat{} and \tmat. The $N \times L$ matrix \tmat{} contains the total number of reads sampled at each locus for each individual. Similarly, \rmat{} is an $N \times L$ matrix containing the number of sampled reads with the reference allele at each locus for each individual. Assuming conditional independence of the sequencing reads given genotypes, the probability distribution for sequencing reads can be factored as

\begin{equation}\label{factored_lik}
P(\bm{R}|\bm{T},\bm{G}, \epsilon) = \displaystyle\prod_{\ell=1}^L\displaystyle\prod_{i=1}^N P(r_{i \ell}|t_{i \ell},g_{i \ell}, \epsilon)\,.
\end{equation}

\noindent For an individual, $i$, at a particular locus, $\ell$, we model the number of sequencing reads containing the reference allele ($r_{i\ell}$) as a Binomial random variable conditional on the total number of sequencing reads ($t_{i\ell} $), the underlying genotype ($g_{i\ell}$) and a constant level of sequencing error ($\epsilon$)

\begin{equation}\label{likelihood}
P(r_{i \ell}|t_{i\ell}, g_{i \ell},\epsilon) = \binom{t_{i \ell}}{r_{i \ell}}
	\begin{cases}
	\epsilon^{r_{i \ell}}(1-\epsilon)^{t_{i \ell}-r_{i \ell}} & \text{if  } g_{i \ell} = 0, \\[0.05in]
	\left(\frac{g_{i \ell}}{\psi}\right)^{r_{i \ell}}\left(1-\frac{g_{i \ell}}{\psi}\right)^{t_{i \ell}-r_{i \ell}} & \text{if  } g_{i \ell} = 1,\ldots,\psi-1, \\[0.05in]
	(1-\epsilon)^{r_{i \ell}}\epsilon^{t_{i \ell}-r_{i \ell}} & \text{if  } g_{i \ell} = \psi\,.
	\end{cases}
\end{equation}


\noindent Since the $r_{i \ell}$'s are the data that we observe, the product of $P(r_{i \ell}|t_{i\ell}, g_{i \ell},\epsilon)$ across loci and individuals will form the likelihood in the model. An important consideration here is that when $g_{i \ell}$ is equal to 0 or $\psi$ (i.e., when the genotype is homozygous) the likelihood will always be 0. To correct for this, we include error ($\epsilon$) into the model. The intuition behind including error is that we may have just not sampled the alternative allele due to lack of sequencing depth or due to a sequencing error. When we do this, the probability distribution for $r_{i\ell}$ given $g_{i\ell}$ and $\epsilon$ is split into $\psi+1$ cases as above in Eq. \ref{likelihood}.
\medskip

The next level in the hierarchy is the conditional prior for genotypes. We assume that the genotypes of the sampled individuals are conditionally independent given the allele frequencies. Factoring the distribution for genotypes and taking the product across loci and individuals gives us the joint probability distribution of genotypes given the ploidy level of the population and the allele frequencies:

\begin{equation}\label{condl_prior}
P(\bm{G}|\psi, \bm{p}) = \displaystyle\prod_{\ell=1}^L\displaystyle\prod_{i=1}^N P(g_{i \ell}|\psi, p_{\ell})\,.
\end{equation}

\noindent We model each $g_{i \ell}$ as a Binomial random variable conditional on the ploidy level of the population and the frequency of the reference allele for locus $\ell$:

\begin{equation*}
P(g_{i \ell}|\psi,p_{\ell}) = \binom{\psi}{g_{i \ell}}\,p_{\ell}^{g_{i \ell}}(1-p_{\ell})^{\psi-g_{i \ell}}\,.
\end{equation*}

The final level of the model is the prior distribution on allele frequencies. Assuming \textit{a priori} independence across loci, we use a Beta distribution with parameters $\alpha$ and $\beta$ both equal to $1$ as our prior distribution for each locus. A Beta(1,1) is equivalent to a Uniform distribution over the interval $[0,1]$, making our choice of prior uninformative. The joint posterior distribution of allele frequencies and genotypes is then equal to the product across all loci and all individuals of the likelihood, the conditional prior on genotypes and the prior distribution on allele frequencies up to a constant of proportionality

\begin{align}\label{posterior}
P(\,\bm{p},\bm{G}|\bm{R},\epsilon) &\propto P(\bm{R}|\bm{T},\bm{G}, \epsilon)P(\bm{G}|\psi,\bm{p})P(\bm{p}) \nonumber \\[0.05in]
&= \displaystyle\prod_{\ell=1}^L\displaystyle\prod_{i=1}^N P(r_{i \ell}|t_{i\ell}, g_{i \ell},\epsilon)P(g_{i \ell}|\psi, p_{\ell})P(p_{\ell})\,.
\end{align}

\noindent The marginal posterior distribution for allele frequencies can be obtained by summing over genotypes

\begin{equation}\label{marg_post_p}
P(\,\bm{p}|\bm{R},\epsilon) \propto \displaystyle\sum_{\bm{G}} P(\,\bm{p},\bm{G}|\bm{R},\epsilon)\,.
\end{equation}

\noindent It would also be possible to examine the marginal posterior distribution of genotypes but for now we will only focus on allele frequencies (see \textbf{Discussion} for potential applications of the marginal distribution of genotypes).

\medskip
\subsection*{Full conditionals and MCMC using Gibbs sampling}
\medskip

\noindent We estimate the joint posterior distribution for allele frequencies and genotypes in Eq. \ref{posterior} using MCMC. This is done using Gibbs sampling of the states $(\,\bm{p},\bm{G})$ in a Markov chain by alternating samples from the full conditional distributions of $\bm{p}$ and $\bm{G}$. Given the setup for our model using Binomial and Beta distributions (which form a conjugate family), analytical solutions for these distributions can be readily acquired \citep{gelman2014bayesian}. The full conditional distribution for allele frequencies is Beta distributed and is given by Eq. \ref{p-full} below:

\begin{equation}\label{p-full}
P(\,p_{\ell}|g_{i \ell},r_{i \ell},\epsilon) = \text{Beta}\left(\alpha= \sum_{i=1}^N g_{i \ell} +1,\; \beta = \sum_{i=1}^N (\psi-g_{i \ell})+1\right),\quad \text{for } \ell = 1,\ldots,L.
\end{equation}

\noindent This full conditional distribution for $p_{\ell}$ has a natural interpretation as it is roughly centered at the proportion of sampled alleles carrying the reference allele divided by the total number of alleles sampled given the current state of $\bm{G}$ in the Markov chain. The ``$+1$'' comes from the prior distribution and won't have a strong influence on the posterior distribution when the sample size is large.
\medskip

The full conditional distribution for genotypes is split into $\psi+1$ cases (similar to the conditional prior), making it a discrete categorical distribution over the possible values for the genotypes $(0,\ldots,\psi)$. Using $k$ as a generic index, the distribution for individual $i$ at locus $\ell$ is

\begin{equation}\label{G-full}
P(g_{i \ell}|g_{(\text{-}i) \ell},p_{\ell},r_{i \ell},\epsilon) = \frac{1}{\mathcal{C}_{i \ell}} \;
	\begin{cases}
	\epsilon^{r_{i \ell}}(1-\epsilon)^{t_{i \ell}-r_{i \ell}}(1-p_{\ell})^\psi & \text{for  } k = 0, \\[0.05in]
	\left(\frac{k}{\psi}\right)^{r_{i \ell}}\left(1-\frac{k}{\psi}\right)^{t_{i \ell}-r_{i \ell}}\displaystyle\binom{\psi}{k}p_{\ell}^{k}\,(1-p_{\ell})^{\psi-k} & \text{for  } k = 1,\ldots,\psi-1, \\[0.05in]
	(1-\epsilon)^{r_{i \ell}}\epsilon^{t_{i \ell}-r_{i \ell}}p_{\ell}^\psi & \text{for  } k = \psi\,,
	\end{cases} 
\end{equation}

\noindent where $g_{(\text{-}i) \ell}$ is the value of the genotypes for all sampled individuals excluding individual $i$ and $\mathcal{C}_{i \ell}$ is a normalizing constant equal to the sum of all of the terms:

\begin{equation*}
\mathcal{C}_{i \ell} = \epsilon^{r_{i \ell}}(1-\epsilon)^{t_{i \ell}-r_{i \ell}}(1-p_{\ell})^\psi + (1-\epsilon)^{r_{i \ell}}\epsilon^{t_{i \ell}-r_{i \ell}}p_{\ell}^\psi + \sum_{k=1}^{\psi-1}\left(\left(\frac{k}{\psi}\right)^{r_{i \ell}}\left(1-\frac{k}{\psi}\right)^{t_{i \ell}-r_{i \ell}}\binom{\psi}{k}\,p_{\ell}^k(1-p_{\ell})^{\psi-k}\right).
\end{equation*}

The conditional distribution for genotypes also has a natural interpretation due to the fact that the next draw for each genotype is based on the probability of each genotype value as determined by the observed sequencing reads multiplied by the probability of drawing the same number of copies of the reference allele based on the current value of $\bm{p}$.

\medskip
\subsection*{Simulation study}
\medskip

Simulations were performed to assess error rates in allele frequency estimation for tetraploid and hexaploid populations. Data were generated under the model by sampling genotypes from a binomial distribution conditional on a fixed, known allele frequency $(\,p_{\ell} = 0.01, 0.05, 0.1, 0.2, 0.4)$. Total read counts were simulated for a single locus using a Poisson distribution with mean coverage ($\lambda$) equal to 5, 10, 20, 50 or 100 reads per individual. We then sampled the number of sequencing reads containing the reference allele from a binomial distribution conditional on the number of total reads, the genotype and sequencing error (Eq. \ref{likelihood}; $\epsilon$ fixed to 0.01). Finally, we varied the number of individuals sampled per population $(N = 5, 10, 20, 30)$ and ran all possible combinations of the simulation settings. Each combination of sequencing coverage, individuals sampled and allele frequency was analyzed using 100 replicates for both tetraploid and hexaploid populations for a total of  20,000 simulation runs. MCMC analyses using Gibbs sampling were run for 50,000 generations with parameter values stored every 50 samples. The first 25\% of the posterior was discarded as burn-in, resulting in 750 posterior samples for each replicate. Convergence on the stationary distribution, $P(\,\bm{p},\bm{G}|\bm{R},\epsilon)$, was assessed by examining trace plots for a subset of runs for each combination of settings and ensuring that the effective sample sizes (ESS) were greater than 200. Deviations from the known underlying allele frequency used to simulate each data set were assessed using the standard deviation of the posterior means of each replicate subtracted from the known value.
\medskip

All simulations were performed using the R statistical package \citep{r2014} on the Oakley cluster at the Ohio Supercomputer Center (\url{https://osc.edu}). Figures were generated using the R packages \textsc{ggplot2} \citep{wickham2009ggplot2}, \textsc{reshape} \citep{wickham2011plyr} and \textsc{plyr} \citep{wickham2007reshape} with additional figure manipulation completed using Inkscape (\url{https://inkscape.org}). MCMC diagnostics were done using the \textsc{coda} package \citep{plummer2006coda}.
\medskip

%%%%%%%%%%%%%%%
\section*{Results}         %%
%%%%%%%%%%%%%%%

\subsection*{Gibbs sampler}
\medskip

Our Gibbs sampling algorithm was able to accurately estimate allele frequencies for a number of simulation settings while simultaneously allowing for genotype uncertainty. Convergence (ESS values > 200) was achieved for all simulation replicates and all trace plots examined also indicated that the Markov chain had reached stationarity (Figure S1). We have aggregated the scripts for our Gibbs sampler as a developmental R package---\textsc{polyfreqs}---which is available on GitHub (\url{https://github.com/pblischak/polyfreqs}). \textsc{polyfreqs} takes advantage of R's ability to incorporate C++ code via the \textsc{Rcpp} package, allowing for a faster implementation of our algorithm \citep{eddelbuettel2011rcpp,eddelbuettel2013rcppBook}.

\medskip
\subsection*{Simulation study}
\medskip

Varying the level of coverage and the number of individuals sampled had the largest effect on the accuracy of allele frequency estimation (Figure 1). Since allele frequencies are a population level parameter, it is not surprising that sampling more individuals from the population leads to better estimates. This appears to be the case even when sequencing coverage is quite low (5x, 10x), which corroborates the observations made by \cite{buerkle2013popModels}. Lower sequence coverage does effect the posterior distribution for allele frequencies and genotypes even when the number of individuals sequenced is large however, by increasing the posterior variance (Figure 2).
\medskip

%%%%%%%%%%%%%%%%%
\section*{Discussion}         %%
%%%%%%%%%%%%%%%%%

The inference of population genetic parameters and the demographic history of polyploid organisms has lagged behind that of diploids for decades. The difficulties associated with these inferences present themselves at two levels. The first of these is the widely known inability to determine the genotypes of polyploids due to ADU. Even though there have been theoretical developments in the description of models for polyploid taxa as early as the 1930s, a large portion of this population genetic theory relies on knowledge about individuals' genotypes \citep[e.g.,][]{haldane1930autopolyploids,wright1938polyploid}. The second complicating factor is the complexity of inheritance patterns and changes in mating systems that accompany WGD events. Polyploid organisms can sometimes mate by both outcrossing or selfing, and can display mixed inheritance patterns at different loci in the genome \citep{dufresne2014polyPopGen}. If genotypes were known, then it might be possible to develop models for dealing with and inferring rates of selfing versus outcrossing, as well as understanding inheritance patterns across the genome. However, ADU only compounds the problems associated with these inferences, making the development and application of appropriate models far more difficult \citep[but see list of software in][]{dufresne2014polyPopGen}. The model we have presented here deals with the first of these two issues by not treating genotypes as observed quantities. Instead, we use the read counts generated by high throughput sequencing platforms as our observed data and integrate across genotype uncertainty when inferring other parameters and thus bypassing the problems caused by ADU.
\medskip

Because our model approximates the joint posterior distribution of allele frequencies and genotypes, it would also be possible to use the marginal posterior distribution of genotypes, $P(\bm{G}|\bm{R},\epsilon)$, to make inferences using existing methods. This could be done using the posterior mode as a maximum \textit{a posteriori} (MAP) estimate of the genotype for downstream analyses, followed by analyzing random samples from the posterior distribution of genotypes. The resulting set of estimates would not constitute a ``true'' posterior distribution of downstream parameters but would allow researchers to interpret their results based on the MAP estimate of the genotypes while still getting a sense for the amount of variation. Other methods for estimating SNP genotypes from high throughput sequencing data include the program \textsc{SuperMASSA}, which models the relative intensity of the two alternative alleles using Normal densities \citep{serang2012supermassa}. 

\medskip
\subsection*{Model adequacy}
\medskip

As noted earlier, the probability model that we use for the inheritance of alleles is one of polysomy without double reduction. In many cases this model may be inappropriate but it can still be informative to check for loci that do or do not follow the model that we assume. Below we describe a simple procedure for rejecting our model of inheritance on a per locus basis using comparisons with the posterior predictive distribution of sequencing reads ($\tilde{\bm{R}}$), which is given by Eq. \ref{post_pred_distr}

\begin{equation}\label{post_pred_distr}
P(\tilde{\bm{R}}|\bm{R},\epsilon) = \int \left(\displaystyle\sum_{\bm{G}} P(\tilde{\bm{R}}|\bm{p},\bm{G}) P(\bm{p},\bm{G}|\bm{R},\epsilon)\right) \text{d}\bm{p}\, .
\end{equation}

\noindent Model checking is an important part of making statistical inferences and can play a role in understanding when a model adequately describes the data being analyzed. In the case of our model, it can serve as a basis for understanding the inheritance patterns of the organism being studied by determining which loci adhere to a simple pattern of polysomic inheritance.
\medskip

Given $M$ posterior samples for the allele frequencies at each locus, $\left\{p_{\ell}^{[1]},p_{\ell}^{[2]},\ldots,p_{\ell}^{[M]} \right\}$, we will simulate new values for the genotypes ($\tilde{g}_{i \ell}$) and reference read counts ($\tilde{r}_{i \ell}$) for all individuals and use the Binomial likelihood (Eq. \ref{likelihood}) as a summary statistic for comparing the observed read count likelihood to the distribution of the predicted read count likelihoods. The use of the likelihood as a summary statistic has been a common practice in comparisons of nucleotide substitution models, as well as models for comparative phylogenetics \citep{ripplinger2010DNAmodels,pennell2015adequacy}. The procedure is as follows:
\medskip

\begin{enumerate}
  \item For locus $\ell = 1,\ldots,L$:
  \begin{enumerate}[label={\arabic{enumi}.\arabic*.}]
    \item For posterior sample $m = 1,\ldots,M$:
    \begin{enumerate}[label={\arabic{enumi}.\arabic{enumi}.\arabic*.}]
      \item Simulate new genotype values $\left( \tilde{g}_{i \ell}^{[m]}\right)$ for all individuals ($i = 1,\ldots,N$) by drawing from a $\text{Binomial}\left( \psi,p_{\ell}^{[m]} \right)$.
      \item Simulate new reference read counts $\left( \tilde{r}_{i \ell}^{[m]} \right)$ from each new genotype for all individuals by drawing from Eq. \ref{likelihood}.
      \item Calculate the Binomial likelihood for sample $m$ as the product across individuals: $\mathcal{L}_{\ell}^{[m]} = \prod_{i=1}^{N}P\left(\tilde{r}_{i \ell}^{[m]} | t_{i \ell},\tilde{g}_{i \ell}^{[m]},\epsilon\right)$.
    \end{enumerate}
    \item Calculate the Binomial likelihood for the observed read counts, $\mathcal{L}_{\ell}^{obs}$, and compare it to the distribution of likelihood values for the predicted reference read counts, $\left\{ \mathcal{L}_{\ell}^{[1]},\ldots,\mathcal{L}_{\ell}^{[M]}\right\}$.
\end{enumerate}
\item Determine if the observed value falls within the 95\% Credible Interval (CI). Repeat for each locus.
\end{enumerate}
\medskip

Observed values of the Binomial likelihood which fall outside of the 95\% CI provide evidence that the locus being examined does not follow a pattern of polysomic inheritance without double reduction. . We implement this posterior predictive model checking procedure in the \textsc{polyfreqs} package with the function \texttt{polyfreqs\_pps}.

\medskip
\subsection*{Extensibility}
\medskip

Talk about how the model can serve as a jumping off point for new models that now that ADU may no longer be as much of an issue. Examples are below.
\medskip

Replace $\psi$ with a vector of values representing the ploidy of each individual $(\bm{\psi} = \{\psi_1,\ldots,\psi_N\})$.
\medskip

%%%%%%%%%%%%%%%%%
\section*{Conclusions}      %%
%%%%%%%%%%%%%%%%%

Our model is one with great potential$\ldots$
\medskip

%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgements}           %%
%%%%%%%%%%%%%%%%%%%%%%%

The authors would like to thank the Ohio Supercomputer Center for access to computing resources and Nick Skomrock for assistance with deriving the full conditional distributions of the model in the diploid case. This manuscript was also greatly improved by comments from the Associate Editor, \textit{editor name}, and \# anonymous reviewers.
\medskip


%%%%%%%%%%%%%%%%%
% References                       %%
%%%%%%%%%%%%%%%%%

\singlespacing

\bibliographystyle{molEcol}
\bibliography{references}

%\doublespacing

%%%%%%%%%%%%%%%%%%%%%%%
\section*{Author Contributions}        %%
%%%%%%%%%%%%%%%%%%%%%%%

Conceived of the study: PDB, LSK and ADW. PDB derived the polyploid model, ran the simulations, coded the R package and wrote the manuscript. PDB, LSK and ADW reviewed all parts of the manuscript and all authors approved of the final version.
\medskip

%%%%%%%%%%%%%%%%%%%%%%
\section*{Data Accessibility}            %%
%%%%%%%%%%%%%%%%%%%%%%

Scripts for simulating the datasets, analyzing them using Gibbs sampling and producing the figures from the resulting output can be all be found on GitHub (\url{https://github.com/pblischak/polyfreqs-ms-data}). We also provide an implementation of the Gibbs sampler for estimating allele frequencies in the developmental R package \textsc{polyfreqs} (\url{https://github.com/pblischak/polyfreqs}). See the package vignette or GitHub wiki for more details (\url{https://github.com/pblischak/polyfreqs/wiki}).
\vfill

%%%%%%%%%%
% Tables          %%
%%%%%%%%%%

\begin{table}
\centering
\rowcolors{1}{white}{gray!25}
\caption{\textbf{Notation and symbols used in the description of the model for estimating allele frequencies in polyploids}. Vector and matrix forms of the variables are also provided when appropriate. }
\vspace{0.2in}
\bgroup
\def\arraystretch{1.45}
\begin{tabular}[l]{l | l}
\hline
\textbf{Symbol} & \textbf{Description}\\ \hline
$L$ & The number of loci. \\
$\ell$ & Index for loci ($\ell\; \in \{1,\ldots,L\}$). \\
$N$ & Total number of individuals sequenced. \\
$i$ & Index for individuals ($i\; \in \{1,\ldots,N\}$). \\
$\psi$ & The ploidy level of individuals in the population (e.g., tetraploid: $\psi$=4). \\
$p_{\ell}$ & Frequency of the reference allele at locus $\ell$. [$\bm{p}$] \\
$g_{i \ell}$ & The number of copies of the reference allele for individual $i$ at locus $\ell$. [$\bm{G}$] \\
$\tilde{g}_{i \ell}$ & Simulated genotype for posterior predictive model checking. \\
$t_{i \ell}$ & The total number of reads for individual $i$ at locus $\ell$. [$\bm{T}$] \\
$r_{i \ell}$ & The number of reads with the reference allele for individual $i$ at locus $\ell$. [$\bm{R}$] \\
$\tilde{r}_{i \ell}$ & Simulated reference read count for posterior predictive model checking. [$\tilde{\bm{R}}$] \\
$\epsilon$ & Sequencing error. \\
\hline
\end{tabular}
\egroup
\label{table1}
\end{table}

%%%%%%%%%%%
% Figures           %%
%%%%%%%%%%%

% Put figures here...

\end{document}