\documentclass[11pt,english,letterpaper,oneside]{article}
\usepackage{amsmath,amssymb,amsfonts,amsthm,enumerate,enumitem}
\usepackage[top=3cm,bottom=3cm,left=2.5cm,right=2.5cm]{geometry}
\usepackage{graphicx,microtype}
\usepackage{textcomp}
\usepackage[T1]{fontenc}
\DisableLigatures[f]{encoding=T1}
\usepackage[skip=2pt]{caption}
\usepackage{lineno}
\usepackage{hyperref}
\linenumbers
\usepackage{setspace,array,float}
\usepackage{bm,upgreek}
\usepackage[compact]{titlesec}
\usepackage[none]{hyphenat}
\setlength{\parskip}{0cm}
\usepackage{natbib}
\setcitestyle{citesep={;},aysep={}}
\usepackage{babel}
\usepackage{datetime}
\pagestyle{plain}
\frenchspacing
\doublespacing
\usepackage[table]{xcolor}
\setcounter{secnumdepth}{0}


\makeatletter
\g@addto@macro\normalsize{%
  \setlength\belowdisplayskip{15pt}
  \setlength\belowdisplayshortskip{10pt}
}
\makeatother

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\newcommand{\tmat}{$\bm{T}$}
\newcommand{\rmat}{$\bm{R}$}
\newcommand{\etal}{\textit{et al}.}
\newdateformat{mydate}{\THEDAY{} \monthname[\THEMONTH], \THEYEAR}
\hfill Version dated: \mydate\today
\vspace{0.25in}

\begin{center}

{\LARGE  \bfseries Accounting for genotype uncertainty in the estimation of allele frequencies in autopolyploids}
\vspace{0.45in}

Paul D. Blischak$^{1,*}$, Laura S. Kubatko$^{1,2}$ and Andrea D. Wolfe$^1$
\vspace{0.45in}


\textit{$^1$Department of Evolution, Ecology and Organismal Biology, Ohio State University,}

\textit{318 W. 12th Avenue, Columbus, OH 43210, USA.}
\bigskip
\bigskip

\textit{$^2$Department of Statistics, Ohio State University,}

\textit{1958 Neil Avenue, Columbus, OH 43210, USA.}


\end{center}
\vspace{0.45in}


\noindent $^*$\textbf{Corresponding author}: Paul Blischak, Ohio State University, Dept. of Evolution, Ecology and Organismal Biology, 318 W. 12th Avenue, Columbus, OH 43210. E-mail: blischak.4@osu.edu.

\vspace{0.45in}

\noindent \textbf{Running title}: Genotype uncertainty in autopolyploids

\vspace{.45in}

%%%%%%%%%%%%%%%%%%%%
\section{Abstract}                      %%
%%%%%%%%%%%%%%%%%%%%

Despite the increasing opportunity to collect large-scale data sets for population genomic analyses, the use of high throughput sequencing to study populations of polyploids has seen little application. This is due in large part to problems associated with determining allele copy number in the genotypes of polyploid individuals (allelic dosage uncertainty--ADU), which complicates the calculation of important quantities such as allele frequencies. Here we describe a statistical model to estimate biallelic SNP frequencies in a population of autopolyploids using high throughput sequencing data in the form of read counts.We bridge the gap from data collection (using restriction enzyme based techniques [e.g., GBS, RADseq]) to allele frequency estimation in a unified inferential framework using a hierarchical Bayesian model to sum over genotype uncertainty. Simulated data sets were generated under various conditions for tetraploid, hexaploid and octoploid populations to evaluate the model's performance and to help guide the collection of empirical data. We also provide an implementation of our model in the R package \textsc{polyfreqs} and demonstrate its use with two example analyses that investigate (i) levels of expected and observed heterozygosity and (ii) model adequacy. Our simulations show that the number of individuals sampled from a population has a greater impact on estimation error than sequencing coverage. The example analyses also show that our model and software can be used to make inferences beyond the estimation of allele frequencies for autopolyploids by providing assessments of model adequacy and estimates of heterozygosity.
\vspace{0.25in}

\noindent (\textbf{Keywords}: allelic dosage uncertainty, genotyping by sequencing, hierarchical Bayesian modeling, polyploidy, population genomics, RADseq)
\vspace{0.25in}

%%%%%%%%%%%%%%%%%%%
\section{Introduction}            %%
%%%%%%%%%%%%%%%%%%%

Biologists have long been fascinated by the occurrence of whole genome duplication (WGD) in natural populations and have recognized its role in the generation of biodiversity \citep{ClausKeckHies1940,StebbinsVariationEvolution,GrantPlantSpeciation,otto2000polyploidy}. Though WGD is thought to have occurred at some point in nearly every major group of eukaryotes, it is a particularly common phenomenon in plants and is regarded by many to be an important factor in plant diversification \citep{wood2009polyploid,soltisd2009diversification,scarpino2014polyploid}. The role of polyploidy in plant evolution was originally considered by some to be a ``dead-end'' \citep{StebbinsVariationEvolution,wagner1970noise,soltisd2014stebbins} but, since its first discovery in the early twentieth century, polyploidy has been continually studied in nearly all areas of botany \citep{winge1917polyploidy,Winkler1916polyploidy,ClausKeckHies1945polyploidy,GrantPlantSpeciation,StebbinsVariationEvolution,soltisD2003polyploid,soltisd2010polyploidUnknowns,soltai2009roleOfHybridization,ramsey2014polEcoProcRoySoc}. Though fewer examples of WGD are currently known for animal systems, groups such as amphibians, fish, and reptiles all exhibit polyploidy \citep{allendorf1984tetraploidFish,gregory2005polyploidyAnimals}. Ancient genome duplications are also thought to have played an important role in the evolution of both plants and animals, occurring in the lineages preceeding the seed plants, angiosperms and vertebrates \citep{ohno1970geneDuplication,otto2000polyploidy,furlong2001animalOctoploid,jiao2011ancientWGD}. These ancient WGD events during the early history of seed plants and angiosperms have been followed by several more WGDs in all major plant groups \citep{cui2006genomeDuplication,scarpino2014polyploid,canon2014polyploidyLegumes}. Recent experimental evidence has also demonstrated increased survivorship and adaptability to foreign environments of polyploid taxa when compared with their lower ploidy relatives \citep{ramsey2011polyploidEcology,Selmecki2015yeastAdaptation}.
\medskip

Polyploids are generally divided into two types based on how they are formed: auto- and allopolyploids. Autopolyploids form when a WGD event occurs within a single evolutionary lineage and typically have polysomic inheritance. Allopolyploids are formed by hybridization between two separately evolving lineages followed by WGD and are thought to have mostly disomic inheritance. Multivalent chromosome pairing during meiosis can occur in allopolyploids, however, resulting in mixed inheritance patterns across loci in the genome [segmental allopolyploids] \citep{StebbinsVariationEvolution}. Autopolyploids can also undergo double reduction, a product of multivalent chromosome pairing wherein segments from sister chromatids move together during meiosis---resulting in allelic inheritance that breaks away from a strict pattern of polysomy \citep{haldane1930autopolyploids}. Autopolyploidy was also thought to be far less common than allopolyploidy, but recent studies have concluded that autopolyploidy occurs much more frequently than originally proposed \citep{soltis2007autopolyploidy,parisod2010autopolyploidy}.
\medskip

The theoretical treatment of population genetic models in polyploids has it origins in the Modern Synthesis with  Fisher, Haldane and Wright each contributing to the development of some of the earliest mathematical models for understanding the genetic patterns of inheritance in polyploids \citep{haldane1930autopolyploids,wright1938polyploid,fisher1943doublereduction}. Early empirical work on polyploids that influenced Fisher, Haldane and Wright include studies on \textit{Lythrum salicaria} by N. Barlow (\citeyear{barlow1913heterostylism}, \citeyear{barlow1923trimorphic}), \textit{Dahlia} by W. J. C. Lawrence (\citeyear{lawrence1929dahlia}) and \textit{Primula} by H. J. Muller (\citeyear{muller1914primula}). The foundation laid down by these early papers has led to the continuing development of population genetic models for polyploids, including models for understanding the rate of loss of genetic diversity and extensions of the coalescent in autotetraploids, as well as modifications of the multispecies coalescent for the inference of species networks containing allotetraploids \citep{moody1993autopolyploids,arnold2012autotetraploidCoal,jones2013allopolyploid}. Much of this progress was described in a review by \cite{dufresne2014polyPopGen}, who outlined the current state of population genetics in polyploids regarding both molecular techniques and statistical models. Not surprisingly, one of the most promising developments for the future of population genetics in polyploids is the advancement of sequencing technologies. A particularly common method of gathering large data sets for genome scale inferences are restriction enzyme based techniques (e.g., RADseq, ddRAD, GBS, etc.), which we will refer to generally as RADseq \citep{miller2007gbs,baird2008radTags,peterson2012ddrad,puritz2014demystifyingRAD}. However, despite its popularity for population genetic inferences at the diploid level, there are many fewer examples of RADseq experiments conducted on polyploid taxa \citep[but see][]{ogden2013sturgeonRADseq,wang2013birchRADseq,logan-young2015polyploidSNP}.
\medskip

Among the primary reasons for the dearth in applying RADseq to polyploids is the issue of allelic dosage uncertainty (ADU), or the inability to fully determine the genotype of a polyploid organism when it is partially heterozygous at a given locus. This is the same problem that has been encountered by other codominant markers such as microsatellites, which have been commonly used for population genetic analyses in polyploids. One way of dealing with allelic dosage that has been used for multi-allelic microsatellite markers has been to code alleles as either present or absent based on electropherogram readings (allelic phenotypes) and to analyze the resulting dominant data using a program such as \textsc{polysat} \citep{clark2007polysat,dufresne2014polyPopGen}. \cite{deSilva2005alleleFreqs} developed a method for inferring allele frequencies using observed allelic phenotype data and used an expectation-maximization algorithm to deal with the incomplete genotype data resulting from ADU. Attempts to directly infer the genotype of polyploid microsatellite loci have also been successfully completed in some cases by using the relative electropherogram peak heights of the alleles in the genotypes \citep{esselink2004polyploidSSR}. The estimation problem would be similar for biallelic SNP data collected using RADseq, where a partially heterozygous polyploid will have high throughput sequencing reads containing both alleles. For a tetraploid, the possible genotypes for a partial heterozygote (alleles A and B) would be AAAB, AABB and ABBB. For a hexaploid they are AAAAAB, AAAABB, AAABBB, AABBBB and ABBBBB. In general, the number of possible genotypes for a biallelic locus of a partially heterozygous $K$-ploid ($K=3,4,5,\ldots$) is $K-1$. A possible solution to this problem for SNPs would be to try to use existing genotype callers and to rely on the relative number of sequencing reads containing the two alleles (similar to what was done for microsatellites). However, this could lead to erroneous inferences when genotypes are simply fixed at point estimates based on read proportions without considering estimation error. Furthermore, when sequencing coverage is low, the number of genotypes that will appear to be equally probable increases with ploidy, making it difficult to distinguish among the possible partially heterozygous genotypes.
\medskip

In this paper we describe a model that aims to address the problems associated with ADU by treating genotypes as a latent variable in a hierarchical Bayesian model and using high throughput sequencing read counts as data. In this way we preserve the uncertainty that is inherent in polyploid genotypes by inferring a probability distribution across all possible values of the genotype, rather than treating them as being directly observed. This approach has been used by \cite{buerkle2013popModels} to deal with uncertainty in calling genotypes in diploids and the work we present here builds off of their earlier models. Our model assumes that the ploidy level of the population is known and that the genotypes of individuals in the population are drawn from a single underlying allele frequency for each locus. These assumptions imply that alleles in the population are undergoing polysomic inheritance without double reduction, which most closely adheres to the inheritance patterns of an autopolyploid. We acknowledge that the model in its current form is an oversimplification of biological reality and realize that it does not apply to a large portion of polyploid taxa. Nevertheless, we believe that accounting for ADU by modeling genotype uncertainty has the potential to be applied more broadly via modifications of the probability model used for the inheritance of alleles, which could lead to more generalized population genetic models for polyploids (see the \textbf{Extensibility} section of the \textbf{Discussion}).
\medskip

%%%%%%%%%%%%%%%%%%%%%%%%
\section{Materials and Methods}        %%
%%%%%%%%%%%%%%%%%%%%%%%%

\noindent Our goal is to estimate the frequency of a reference allele for each locus sampled from a population of known ploidy ($\psi$), where the reference allele can be chosen arbitrarily between the two alleles at a given biallelic SNP. To do this we extend the population genomic models of \cite{buerkle2013popModels}, which employ a Bayesian framework to model high throughput sequencing reads ($\bm{T},\bm{R}$), genotypes ($\bm{G}$) and allele frequencies ($\bm{p}$), to the case of arbitrary ploidy. The idea behind the model is to view the sequencing reads gathered for an individual as a random sample from the unobserved genotype at each locus. Genotypes can then be treated as a parameter in a probability model that governs how likely it is that we see a particular number of sequencing reads carrying the reference allele. Similarly, we can treat genotypes as a random sample from the underlying allele frequency in the population (assuming Hardy-Weinberg equilibrium). For our model, a genotype is simply a count of the number of reference alleles at a locus which can range from 0 (a homozygote with no reference alleles in the genotype) to $\psi$ (a homozygote with only reference alleles in the genotype). All whole numbers in between 0 and $\psi$ represent partially heterozygous genotypes. This hierarchical setup addresses the problems associated with ADU by treating genotypes as a latent variable that can be integrated out using Markov chain Monte Carlo (MCMC).

\medskip
\subsection{Model setup} %%%%%%%%%%%%%%%%%%%%%%%%%%
\medskip

Here we consider a sample of $N$ individuals from a single population of ploidy level $\psi$ sequenced at $L$ unlinked SNPs. The data for the model consist of two matrices containing counts of high throughput sequencing reads mapping to each locus for each individual: \rmat{} and \tmat. The $N \times L$ matrix \tmat{} contains the total number of reads sampled at each locus for each individual. Similarly, \rmat{} is an $N \times L$ matrix containing the number of sampled reads with the reference allele at each locus for each individual. Then for individual $i$ at locus $\ell$, we model the number of sequencing reads containing the reference allele ($r_{i\ell}$) as a Binomial random variable conditional on the total number of sequencing reads ($t_{i\ell} $), the underlying genotype ($g_{i\ell}$) and a constant level of sequencing error ($\epsilon$)

\begin{equation}\label{likelihood}
P(r_{i \ell}|t_{i\ell}, g_{i \ell},\epsilon) = \binom{t_{i \ell}}{r_{i \ell}} g_\epsilon^{r_{i \ell}}(1-g_\epsilon)^{t_{i \ell}-r_{i \ell}}\,.
\end{equation}

\noindent Here $g_\epsilon$ is the probability of observing a read containing the reference allele corrected for sequencing error

\begin{equation}\label{g_error}
	g_\epsilon = \left(\frac{g_{i \ell}}{\psi}\right)(1-\epsilon) + \left(1-\frac{g_{i \ell}}{\psi}\right)\epsilon \,.
\end{equation}

\noindent The intuition behind including error is that we want to calculate the probability that we observe a read containing the reference allele. There are two ways that this can happen. (1) Reads are drawn from the reference allele(s) in the genotype with probability $\frac{g_{i\ell}}{\psi}$ but are only observed as reference reads if they are not errors (probability $1-\epsilon$). (2) Similarly, reads from the non-reference allele(s) in the genotype are drawn with probability $1-\frac{g_{i\ell}}{\psi}$ but can be mistakenly read as a coming from a reference allele if an error occurs (probability $\epsilon$). The sum across these two possibilities gives the overall probability of observing a read containing the reference allele. If we also assume conditional independence of the sequencing reads given the genotypes, the joint probability distribution for sequencing reads is given by

\begin{equation}\label{factored_lik}
P(\bm{R}|\bm{T},\bm{G}, \epsilon) = \displaystyle\prod_{\ell=1}^L\displaystyle\prod_{i=1}^N P(r_{i \ell}|t_{i \ell},g_{i \ell}, \epsilon)\,.
\end{equation}

\noindent Since the $r_{i \ell}$'s are the data that we observe, the product of $P(r_{i \ell}|t_{i\ell}, g_{i \ell},\epsilon)$ across loci and individuals will form the likelihood in the model.
\medskip

The next level in the hierarchy is the conditional prior for genotypes. We model each $g_{i \ell}$ as a Binomial random variable conditional on the ploidy level of the population and the frequency of the reference allele for locus $\ell$ ($p_{\ell}$):

\begin{equation*}
P(g_{i \ell}|\psi,p_{\ell}) = \binom{\psi}{g_{i \ell}}\,p_{\ell}^{\,g_{i \ell}}(1-p_{\ell})^{\psi-g_{i \ell}}\,.
\end{equation*}

\noindent We also assume that the genotypes of the sampled individuals are conditionally independent given the allele frequencies, which is equivalent to taking a random sample from a population in Hardy-Weinberg equilibrium. Factoring the distribution for genotypes and taking the product across loci and individuals gives us the joint probability distribution of genotypes given the ploidy level of the population and the vector of allele frequencies at each locus ($\bm{p}=\{p_1,\ldots,p_L\}$):

\begin{equation}\label{condl_prior}
P(\bm{G}|\psi, \bm{p}) = \displaystyle\prod_{\ell=1}^L\displaystyle\prod_{i=1}^N P(g_{i \ell}|\psi, p_{\ell})\,.
\end{equation}

\noindent We choose here to ignore other factors that may be influencing the distribution of genotypes such as double reduction. In general, double reduction will act to increase homozygosity \citep{hardy2015autopolyploids}. However, it is more prevalent for loci that are farther away from the centromere, which makes the estimation of a global double reduction parameter (typically denoted $\alpha$) inappropriate for the thousands of loci gathered from across the genome using techniques such as RADseq. It might be possible to estimate a per locus rate of double reduction ($\alpha_{\ell}$) but this would add an additional parameter that would need to be estimated for each locus, perhaps unnecessarily if the majority end up being equal, or close, to 0.
\medskip

The final level of the model is the prior distribution on allele frequencies. Assuming \textit{a priori} independence across loci, we use a Beta distribution with parameters $\alpha$ and $\beta$ both equal to $1$ as our prior distribution for each locus. A Beta(1,1) is equivalent to a Uniform distribution over the interval $[0,1]$, making our choice of prior uninformative. The joint posterior distribution of allele frequencies and genotypes is then equal to the product across all loci and all individuals of the likelihood, the conditional prior on genotypes and the prior distribution on allele frequencies up to a constant of proportionality

\begin{align}\label{posterior}
P(\,\bm{p},\bm{G}|\bm{T}, \bm{R},\epsilon) &\propto P(\bm{R}|\bm{T},\bm{G}, \epsilon)P(\bm{G}|\psi,\bm{p})P(\bm{p}) \nonumber \\[0.05in]
&= \displaystyle\prod_{\ell=1}^L\displaystyle\prod_{i=1}^N P(r_{i \ell}|t_{i\ell}, g_{i \ell},\epsilon)P(g_{i \ell}|\psi, p_{\ell})P(p_{\ell})\,.
\end{align}

\noindent The marginal posterior distribution for allele frequencies can be obtained by summing over genotypes

\begin{equation}\label{marg_post_p}
P(\,\bm{p}|\bm{T}, \bm{R},\epsilon) \propto \displaystyle\sum_{\bm{G}} P(\,\bm{p},\bm{G}|\bm{T}, \bm{R},\epsilon)\,.
\end{equation}

\noindent It would also be possible to examine the marginal posterior distribution of genotypes but here we will focus primarily on allele frequencies.

\medskip
\subsection{Full conditionals and MCMC using Gibbs sampling} %%%%%%%%%%%%%%%%%%%%%%%%
\medskip

\noindent We estimate the joint posterior distribution for allele frequencies and genotypes in Eq. \ref{posterior} using MCMC. This is done using Gibbs sampling of the states $(\,\bm{p},\bm{G})$ in a Markov chain by alternating samples from the full conditional distributions of $\bm{p}$ and $\bm{G}$. Given the setup for our model using Binomial and Beta distributions (which form a conjugate family), analytical solutions for these distributions can be readily acquired \citep{gelman2014bayesian}. The full conditional distribution for allele frequencies is Beta distributed and is given by Eq. \ref{p-full} below:

\begin{equation}\label{p-full}
p_{\ell}\,|\,g_{i \ell},r_{i \ell},\epsilon \: \sim \: \text{Beta}\left(\alpha= \sum_{i=1}^N g_{i \ell} +1,\; \beta = \sum_{i=1}^N (\psi-g_{i \ell})+1\right),\quad \text{for } \ell = 1,\ldots,L.
\end{equation}

\noindent This full conditional distribution for $p_{\ell}$ has a natural interpretation as it is roughly centered at the proportion of sampled alleles carrying the reference allele divided by the total number of alleles sampled. The ``$+1$'' comes from the prior distribution and will not have a strong influence on the posterior when the sample size is large.
\medskip

The full conditional distribution for genotypes is a discrete categorical distribution over the possible values for the genotypes $(0,\ldots,\psi)$. The distribution for individual $i$ at locus $\ell$ is

\begin{equation}\label{G-full}
P(g_{i \ell}|g_{(\text{-}i) \ell},p_{\ell},r_{i \ell},\epsilon) = \binom{t_{i\ell}}{r_{i\ell}} g_\epsilon^{r_{i \ell}}(1-g_\epsilon)^{t_{i \ell}-r_{i \ell}}\displaystyle\binom{\psi}{g_{i \ell}}p_{\ell}^{\,g_{i \ell}}\,(1-p_{\ell})^{\psi-g_{i \ell}}\,,
\end{equation}

\noindent where $g_{(\text{-}i) \ell}$ is the value of the genotypes for all sampled individuals excluding individual $i$ and $g_\epsilon$ is the same as Eq. \ref{g_error}. The full conditional distribution for genotypes can be seen as the product of two quantities: (1) the probability of each of the possible genotypes based on the observed  reference reads and (2) the probability of drawing each genotype given the allele frequency for that locus in the population.
\medskip  

We begin our Gibbs sampling algorithm in a random position in parameter space through the use of uniform probability distributions. The genotype matrix is initialized with random draws from a Discrete Uniform distribution ranging from $0$ to $\psi$ and the initial allele frequencies are drawn from a Uniform distribution on the interval [0,1].

\medskip
\subsection{Simulation study} %%%%%%%%%%%%%%%%%%%%%%%
\medskip

Simulations were performed to assess error rates in allele frequency estimation for tetraploid, hexaploid and octoploid populations ($\psi$ = 4, 6 and 8, respectively). Data were generated under the model by sampling genotypes from a Binomial distribution conditional on a fixed, known allele frequency $(\,p_{\ell} = 0.01, 0.05, 0.1, 0.2, 0.4)$. Total read counts were simulated for a single locus using a Poisson distribution with mean coverage equal to 5, 10, 20, 50 or 100 reads per individual. We then sampled the number of sequencing reads containing the reference allele from a Binomial distribution conditional on the number of total reads, the genotype and sequencing error (Eq. \ref{likelihood}; $\epsilon$ fixed to 0.01). Finally, we varied the number of individuals sampled per population $(N = 5, 10, 20, 30)$ and ran all possible combinations of the simulation settings. Our choice for the number of individuals to simulate was intended to reflect sampling within a \textit{single} population/locality and not that of an entire population genetics study. Furthermore, RAD sequencing is used at various taxonomic levels from population genetics to phylogenetics \citep[e.g.,][]{rheindt2013zimmerius,eaton2015oaks}, and we wanted our simulations to be informative across these applications. Each combination of sequencing coverage, individuals sampled and allele frequency was analyzed using 100 replicates for tetraploid, hexaploid and octoploid populations for a total of  30,000 simulation runs. MCMC analyses using Gibbs sampling were run for 100,000 generations with parameter values stored every 100th generation. The first 25\% of the sample was discarded as burn-in, resulting in 750 posterior samples for each replicate. Convergence on the stationary distribution, $P(\,\bm{p},\bm{G}|\bm{R},\epsilon)$, was assessed by examining trace plots for a subset of runs for each combination of settings and ensuring that the effective sample sizes (ESS) were greater than 200. Deviations from the known underlying allele frequency used to simulate each data set were assessed by taking the posterior mean of each replicate and calculating the root mean squared error (RMSE) based on the true underlying value. We also compared the posterior mean as an estimate of the allele frequency at a locus to a more simple estimate calculated directly from the read counts (mean read ratio): $\frac{1}{N}\sum_i\frac{r_{i\ell}}{t_{i\ell}}$. Comparisons between estimates were again made using the RMSE.
\medskip

All simulations were performed using the R statistical programming language \citep{r2014} on the Oakley cluster at the Ohio Supercomputer Center (\url{https://osc.edu}). Figures were generated using the R packages \textsc{ggplot2} \citep{wickham2009ggplot2} and \textsc{reshape} \citep{wickham2007reshape}, with additional figure manipulation completed using Inkscape (\url{https://inkscape.org}). MCMC diagnostics were done using the \textsc{coda} package \citep{plummer2006coda}. All scripts are available on GitHub (\url{https://github.com/pblischak/polyfreqs-ms-data}) in the \texttt{`code/'} folder and all simulated data sets are in the \texttt{`raw\_data/'} folder.
\medskip

\subsection{Example analyses of autotetraploid potato (\textit{Solanum tuberosum})} %%%%%%%%%%%%%%%%%%%%%%%%%%%
\medskip

To further evaluate the model and to demonstrate its use we present an example analysis using an empirical data set collected for autotetraploid potato (\textit{Solanum tuberosum}) using the Illumina GoldenGate platform \citep{anithakumari2010goldenGate,voorrips2011fitTetra}. Though these data aren't the typical reads returned by RADseq experiments, they still represent the same type of binary response data that our model uses to get a probability distribution for biallelic SNP genotypes. A detailed walkthrough with the code used for each step is provided as Supplemental Material. The data set and output are also available on GitHub (\url{https://github.com/pblischak/polyfreqs-ms-data}) in the \texttt{`example/'} folder.
\medskip

\subsubsection{\itshape Calculating expected and observed heterozygosity}
\medskip

One advantage of using a Bayesian framework for our model is that we can approximate a posterior distribution for any quantity that is a functional transformation of the parameters that we are estimating without doing any additional MCMC simulation \citep{gelman2014bayesian}. Two such quantities that are often used in population genetics are the observed and expected heterozygosity, which are in turn used for calculating the various fixation indices ($F_{IS}$, $F_{IT}$, $F_{ST}$) introduced by \cite{wright1951Fstats}. To analyze levels of heterozygosity in this way, we used the estimators of \cite{hardy2015autopolyploids} to calculate the per locus observed ($\mathcal{H}_o$) and expected ($\mathcal{H}_e$) heterozygosity for each stored sample of the joint posterior distribution in Eq. \ref{posterior}. This procedure is especially useful because it estimates heterozygosity while taking into account ADU by utilizing the marginal posterior distribution of genotypes. Given a total of $M$ posterior samples of genotypes and allele frequencies, we calculate the $m^\text{th}$ ($m=1,\dots,M$) estimate of the observed heterozygosity using Eq. \ref{het-obs} [numerator of Eq. 7 in \cite{hardy2015autopolyploids}]:

\begin{equation}\label{het-obs}
\mathcal{H}^{[m]}_o = \frac{1}{N} \sum_i h_{i}^{[m]} = \frac{1}{N} \sum_i \frac{g_{i\ell}^{[m]}(\psi-g_{i\ell}^{[m]})}{\binom{\psi}{2}}\, .
\end{equation}

\noindent Similarly, the $m^\text{th}$ estimate of the expected heterozygosity is calculated using Eq. \ref{het-exp} [denominator of Eq. 8 in \cite{hardy2015autopolyploids}]:

\begin{equation}\label{het-exp}
\mathcal{H}^{[m]}_e = \frac{N}{N-1} \left[1 - (p_{\ell}^{[m]})^2 - (1-p_{\ell}^{[m]})^2 - \frac{\psi-1}{\psi N^2}\sum_i h_{i}^{[m]}\right]\,.
\end{equation}

\noindent The posterior distribution of a multi-locus estimate of heterozygosity can then be approximated by taking the average across loci for each of the per locus posterior samples.
\medskip

To evaluate levels of heterozygosity in autotetraploid potato, we obtained biallelic count data for 224 accessions collected at 384 loci using the Illumina GoldenGate platform from the R package \textsc{fitTetra} \citep{voorrips2011fitTetra}, which provides the data set as part of the package. We chose the `X' reading to be the count data for the reference allele and added the `X' and `Y' readings together to get the total read counts (`X' and `Y' represent the counts of the two alternative alleles). Initial attempts to analyze the data set using our Gibbs sampling algorithm were unsuccessful due to arithmetic underflow. This was due to the fact that the counts/intensities returned by the Illumina GoldenGate platform are on a different scale ($\sim$10,000-20,000+) than the read counts that would be expected from a RADseq experiment. To alleviate this problem, we rescaled the data set while preserving the relative dosage information by dividing the GoldenGate count readings by 100 and rounding to the nearest whole number. We then analyzed the rescaled count data using 100,000 MCMC generations, sampling every 100 generations and using the stored samples of the allele frequencies and genotypes to calculate the observed and expected heterozygosity for a total of 1,000 posterior samples of the per locus observed and expected heterozygosity. We also compared post burn-in (25\%) allele frequency estimates based on the posterior mean to the simple allele frequency estimate based directly on read counts used previously (mean read ratio). Posterior distributions for multi-locus estimates of observed and expected heterozygosity were obtained by taking the average across loci for each posterior sample of the per locus estimates using a burn-in of 25\%.
\medskip

\subsubsection{{\itshape Evaluating model adequacy}}
\medskip

As noted earlier, the probability model that we use for the inheritance of alleles is one of polysomy without double reduction. In some cases, this model may be inappropriate. Therefore, it can be informative to check for loci that do not follow the model that we assume. Below we describe a procedure for rejecting our model of inheritance on a per locus basis using comparisons with the posterior predictive distribution of sequencing reads. Model checking is an important part of making statistical inferences and can play a role in understanding when a model adequately describes the data being analyzed. In the case of our model, it can serve as a basis for understanding the inheritance patterns of the organism being studied by determining which loci adhere to a simple pattern of polysomic inheritance. Other sources of disequilibrium that could indicate poor model fit include inbreeding, null alleles and allele drop out \citep[\textit{sensu}][]{arnold2013RADseq}, making this posterior predictive model check more broadly applicable for RADseq data.
\medskip

Given $M$ posterior samples for the allele frequencies at locus $\ell$, $\left\{p_{\ell}^{[1]},p_{\ell}^{[2]},\ldots,p_{\ell}^{[M]} \right\}$, we simulate new values for the genotypes ($\tilde{g}_{i \ell}$) and reference read counts ($\tilde{r}_{i \ell}$) for all individuals and use the ratio of simulated reference read counts to observed total read counts $\left( \frac{\tilde{r}_{i \ell}}{t_{i \ell}} \right) $ as a summary statistic for comparing the observed read count ratios to the distribution of the predicted read count ratios. The use of the likelihood (or similar quantities) as a summary statistic has been a common practice in posterior predictive comparisons of nucleotide substitution models, and more recently for comparative phylogenetics \citep{ripplinger2010DNAmodels,reid2014poorfit,pennell2015adequacy}. We use the ratio of reference to total read counts here because it is the maximum likelihood estimate of the probability of success for a Binomial random variable and because it is a simple quantity to calculate. The use of other summary statistics, or a combination of multiple summary statistics, would also possible. The procedure for our posterior predictive model check is as follows:
\medskip

\begin{enumerate}
  \item For locus $\ell = 1,\ldots,L$:
  \begin{enumerate}[label={\arabic{enumi}.\arabic*.}]
    \item For posterior sample $m = 1,\ldots,M$:
    \begin{enumerate}[label={\arabic{enumi}.\arabic{enumi}.\arabic*.}]
      \item Simulate new genotype values $\left( \tilde{g}_{i \ell}^{[m]}\right)$ for all individuals ($i = 1,\ldots,N$) by drawing from a $\text{Binomial}\left( \psi,p_{\ell}^{[m]} \right)$.
      \item Simulate new reference read counts $\left( \tilde{r}_{i \ell}^{[m]} \right)$ from each new genotype for all individuals by drawing from Eq. \ref{likelihood}.
      \item Calculate the reference read ratio for the simulated data for sample $m$ and sum across individuals: $\mathcal{\tilde{S}}_{\ell}^{[m]} = \sum_{i=1}^{N} \left(\frac{\tilde{r}_{i \ell}^{[m]}}{t_{i \ell}} \right)$.
      \item Calculate the reference read ratio for the observed data and sum across individuals: $\mathcal{S}_{\ell} = \sum_{i=1}^{N} \left(\frac{r_{i \ell}}{t_{i \ell}} \right)$.
    \end{enumerate}
    \item Calculate the difference between the observed reference read ratio and the $M$ simulated reference read ratios: $\left\{ \mathcal{S}_{\ell}-\mathcal{\tilde{S}}_{\ell}^{[1]},\ldots,\mathcal{S}_{\ell}-\mathcal{\tilde{S}}_{\ell}^{[M]}\right\}$.
\end{enumerate}
\item Determine if the 95\% highest posterior density (HPD) interval of the distribution of re-centered reference read ratios contains 0.
\end{enumerate}
\medskip

When the distribution of the differences in ratios between the observed and simulated data sets does not contain 0 in the 95\% HPD interval, it provides evidence that the locus being examined does not follow a pattern of strict polysomic inheritance. A similar approach could be used on an individual basis by comparing the observed ratio of reference reads to the predicted ratios for each individual at each locus. We used this posterior predictive model checking procedure to assess model adequacy in the potato data set using the posterior distribution of allele frequencies estimated in the previous section with 25\% of the samples discarded as burn-in. 
\medskip

%%%%%%%%%%%%%%%
\section{Results}         %%
%%%%%%%%%%%%%%%

Our Gibbs sampling algorithm was able to accurately estimate allele frequencies for a number of simulation settings while simultaneously allowing for genotype uncertainty. There were no indications of a lack of convergence (ESS values > 200) for any of the simulation replicates and all trace plots examined also indicated that the Markov chain had reached stationarity. Running the MCMC for 100,000 generations and sampling every 100th generation appeared to be suitable for our analyses and we recommend it as a starting point for running most data sets. Reducing the number of generations and sampling more frequently (e.g., 50,000 generations sampled every 50 generations) could be a potential work around for larger data sets. When doing test runs we went as low as 20,000 generations sampled every 20th generation, which still passed our diagnostic tests for convergence. This is likely because the parameter space of our model is not overly difficult to navigate so stationarity is reached rather quickly. Ultimately, the deciding factor on how long to run the analysis and how frequently to sample the chain will come down to assessing convergence.

\medskip
\subsection{Simulation study}
\medskip

Increasing the number of individuals sampled had the largest effect on the accuracy of allele frequency estimation (Figure \ref{fig1:rmse}). Since allele frequencies are population parameters, it is not surprising that sampling more individuals from the population leads to better estimates. This appears to be the case even when sequencing coverage is quite low (5x, 10x), which corroborates the observations made by \cite{buerkle2013popModels}. This is not to say, however, that sequencing coverage has no effect on the posterior distribution of allele frequencies. Lower sequencing coverage affects the posterior distribution by increasing the posterior standard deviation (Figure \ref{fig2:coverage-sd}). An interesting pattern that emerged during the simulation study is the observation that the allele frequencies closer to 0.5 tend to have higher error rates, which is to be expected given that the variance of a Binomial random variable is highest when the probability of success is 0.5. We also observed small differences in the RMSE between ploidy levels, with estimates increasing in accuracy with increasing ploidy. Comparisons between the posterior mean and mean read ratio estimates of allele frequencies (Figure S1) show that the estimate based on read ratios has a lower RMSE than the posterior mean when the true allele frequency is low ($p_\ell=0.01, 0.05$) but has higher error rates than the posterior mean for allele frequencies closer to 0.5. When sequencing coverage is greater than 10x and the number of individuals sampled is greater than 20, the two estimates are almost indistinguishable.

\medskip
\subsection{Example analyses}
\medskip

Our analyses of \textit{Solanum tuberosum} tetraploids showed levels of heterozygosity consistent with a pattern of excess outbreeding ($\mathcal{H}_o > \mathcal{H}_e$). In fact, the posterior distributions of the multi-locus estimates of observed and expected heterozygosity do not overlap at all (Figure \ref{fig3:het}). The assessment of model adequacy also showed that 49 out of the 384 loci ($\sim$13\%) were a poor fit to the model of polysomic inheritance that we assume. The allele frequency estimates using the posterior mean and the mean read ratio provided similar estimates and were comparable for most loci. For loci in which the frequency of the reference allele is very low, the read ratio estimate tends to be higher than the posterior mean. However, the overall pattern does not indicate over or under estimation for most allele frequencies (Figure S2). When we took the difference between the estimates at each locus, the distribution was centered near 0 (Figure S3).
\medskip

%%%%%%%%%%%%%%%%%
\section{Discussion}         %%
%%%%%%%%%%%%%%%%%

The inference of population genetic parameters and the demographic history of non-model polyploid organisms has consistently lagged behind that of diploids. The difficulties associated with these inferences present themselves at two levels. The first of these is the widely known inability to determine the genotypes of polyploids due to ADU. Even though there have been theoretical developments in the description of models for polyploid taxa as early as the 1930s, a large portion of this population genetic theory relies on knowledge about individuals' genotypes \citep[e.g.,][]{haldane1930autopolyploids,wright1938polyploid}. The second complicating factor is the complexity of inheritance patterns and changes in mating systems that often accompany WGD events. Polyploid organisms can sometimes mate by both outcrossing or selfing, and can display mixed inheritance patterns at different loci in the genome \citep{dufresne2014polyPopGen}. If genotypes were known, then it might be easier to develop and test models for dealing with and inferring rates of selfing versus outcrossing, as well as understanding inheritance patterns across the genome. However, ADU only compounds the problems associated with these inferences, making the development and application of appropriate models far more difficult \citep[but see list of software in][]{dufresne2014polyPopGen}. The model we have presented here deals with the first of these two issues by not treating genotypes as observed quantities. Almost all other methods of genotype estimation for polyploids treat the genotype as the primary parameter of interest. Our model is different in that we still use the read counts generated by high throughput sequencing platforms as our observed data but instead integrate across genotype uncertainty when inferring other parameters, thus bypassing the problems caused by ADU.
\medskip

Despite our focus on bypassing ADU, an important consideration for the model we present here is that, because it approximates the joint posterior distribution of allele frequencies and genotypes, it would also be possible to use the marginal posterior distribution of genotypes to make inferences using existing methods. This could be done using the posterior mode as a maximum \textit{a posteriori} (MAP) estimate of the genotype for downstream analyses, followed by analyzing the samples taken from the marginal posterior distribution of genotypes. The resulting set of estimates would not constitute a ``true'' posterior distribution of downstream parameters but would allow researchers to interpret their results based on the MAP estimate of the genotypes while still getting a sense for the amount of variation in their estimates. Using the marginal posterior distribution of genotypes in this way could technically be applied to any type of polyploid, but is only really appropriate for autopolyploids due to the model of inheritance that is used. Other methods for estimating SNP genotypes from high throughput sequencing data include the program \textsc{SuperMASSA}, which models the relative intensity of the two alternative alleles using Normal densities \citep{serang2012supermassa}.
\medskip

A second important factor for using our model is that, although estimates of allele frequencies can be accurate when sequencing coverage is low and sample sizes are large (see Figure S4 for a direct comparison between sample size and coverage), the resulting distribution for genotypes is likely going to be quite diffuse. For analyses that treat genotypes as a nuisance parameter, this is not an issue since we can integrate across genotype uncertainty. However, if the genotype \textit{is} of primary interest, then the experimental design of the study will need to change to acquire higher coverage at each locus for more accurate genotype estimation. Therefore, the decision between sequencing more individuals with lower average coverage versus sequencing fewer individuals with higher average coverage depends primarily on whether the genotypes will be used or not.

\medskip
\subsection{Extensibility}
\medskip

The modular nature of our hierarchical model can allow for the addition and modification of levels in the hierarchy. One of the simplest extensions to the model that can build directly on the current setup would be to consider loci with more than two alleles. This can be done using Multinomial distributions for sequencing reads and genotypes and a Dirichlet prior on allele frequencies \citep[the Multinomial and Dirichlet distributions form a conjugate family;][]{gelman2014bayesian}. We could also model populations of mixed ploidy by using a vector of individually assigned ploidy levels instead of assuming a single value for the whole population $(\bm{\psi} = \{\psi_1,\ldots,\psi_N\})$. However, this would assume random mating among ploidy levels.
\medskip


\subsubsection{{\itshape Double reduction}}
\medskip

The inclusion of double reduction into the model is a difficult consideration for genome wide data collected using high throughput sequencing platforms. The number of parameters estimated by our model is $L\times(N+1)$ and including double reduction would add an additional $L$ parameters, bringing the total to $L\times(N+2)$. Though the addition of these parameters would not prohibit an analysis using Gibbs sampling, we chose to implement the simpler equilibrium model. We hope to include double reduction in future models but feel that our posterior predictive model checking procedure will prove sufficient for identifying loci in disequilibrium with our current implementation. Another concern that we had regarding double reduction is that it can be confounded with the overall signal of inbreeding, making it especially difficult to tease apart the specific effects of double reduction alone \citep{hardy2015autopolyploids}. However, because the probability of double reduction at a locus ($\alpha_\ell$) depends on its distance from the centromere (call it $x_\ell$), a potential way to estimate $\alpha_\ell$ would be to use the $x_\ell$'s as predictor variables in a linear model: $\alpha_{\ell} = \beta_0 + \beta_1 x_{\ell}$. This would only add two additional parameters ($\beta_0$ and $\beta_1$) that would need to be estimated and would be completely independent of the number of loci analyzed. The downside to this approach is that it would only be applicable for polyploid organisms with sequenced genomes (or the genome of a diploid progenitor), making the use of such a model impractical for the time being.
\medskip

\subsubsection{{\itshape Additional levels in the hierarchical model}}
\medskip

The place where we believe our model could have the greatest impact is through modifications and extensions of the probability model used for the inheritance of alleles. These models have been difficult to apply in the past as a result of genotype uncertainty. However, using our model as a starting point, it could be possible to infer patterns of inheritance (polysomy, disomy, heterosomy) and other demographic parameters (e.g., effective population size, population differentiation) without requiring direct knowledge about the genotypes of the individuals in the population. For example, Haldane's (\citeyear{haldane1930autopolyploids}) model of genotype frequencies for autopolyploids that are partially selfing could be used to infer the prevalence of self-fertilization within a population. Another possible approach would be to use general disequilibrium coefficients ($D_A$) to model departures from Hardy-Weinberg equilibrium \citep{hernandez1989disequilibrium,weir1996GeneticDataAnalysisII}. A more recent model described by \cite{stift2008polyploidInheritance} used microsatellites to infer the different inheritance patterns (disomic, tetrasomic, intermediate) for tetraploids in the genus \textit{Rorippa} (Brassicaceae) following crossing experiments. The reformulation of such a model for biallelic SNPs gathered using high throughput sequencing could provide a suitable framework for understanding inheritance patterns across the genome. An ideal model would be one that could help to understand genome-wide inheritance patterns for a polyploid of arbitrary formation pathway (autopolyploid $\leftrightarrow$ allopolyploid) without the need conduct additional experiments. However, to our knowledge, such a model does not currently exist. 
\medskip

%%%%%%%%%%%%%%%%%
\section{Conclusions}      %%
%%%%%%%%%%%%%%%%%

The recent emergence of models for genotype uncertainty in diploids has introduced a theoretical framework for dealing with the fact that genotypes are unobserved quantities \citep{gompert2012bgc,buerkle2013popModels}. Our extension of this theory to cases of higher ploidy (specifically to autopolyploids) progresses naturally from the original work but also serves to alleviate the deeper issue of ADU. The power and flexibility of these models as applied at the diploid level has the potential to be replicated for polyploid organisms with the addition of suitable models for allelic inheritance. The construction of hierarchical models containing probability models for ADU, allelic inheritance and perhaps even additional levels for important parameters such as F-statistics or the allele frequency spectrum also have the potential to provide key insights into the population genetics of polyploids \citep{gompert2011bamova,buerkle2013popModels}. Future work on such models will help to progress the study of polyploid taxa and could eventually lead to more generalized models for understanding the processes that have shaped their evolutionary histories.
\medskip

%%%%%%%%%%%%%%%%
\section{Software note}  %%
%%%%%%%%%%%%%%%%

We have combined the scripts for our Gibbs sampler as an R package---\textsc{polyfreqs}---which is available on GitHub (\url{https://github.com/pblischak/polyfreqs}). Though \textsc{polyfreqs} is written in R, it deals with the large data sets that are generated by high throughput sequencing platforms in two ways. First, it takes advantage of R's ability to incorporate C++ code via the \textsc{Rcpp} and \textsc{RcppArmadillo} packages, allowing for a faster implementation of our MCMC algorithm \citep{eddelbuettel2011rcpp,eddelbuettel2013rcppBook,eddelbuettel2014rcpparmadillo}. Second, since the model assumes independence between loci, \textsc{polyfreqs} can facilitate the process of parallelizing analyses by splitting the total read count and reference read count matrices into subsets of loci which can be analyzed at the same time on separate nodes of a computing cluster. Additional features of the program include:
\medskip

\begin{itemize}
	\item Estimation of posterior distributions of per locus observed and expected heterozygosity (\texttt{het\_obs} and \texttt{het\_exp}, respectively).
	\item Maximum \textit{a posteriori} (posterior mode) estimation of genotypes using the \texttt{get\_map\_genotypes()} function.
	\item Posterior predictive model checking using the \texttt{polyfreqs\_pps()} function.
	\item Simulation of high throughput sequencing read counts and genotypes from user specified allele frequencies using the \texttt{sim\_reads()} function.
	\item Options for controlling program output such as writing genotype samples to file, printing MCMC updates to the R console, etc.
	\item Simple input format using tab delimited text files that can be directly imported into R using the \texttt{read.table()} function. The format is as follows:
	\begin{enumerate}
		\item An optional row of locus names (use \texttt{header=TRUE} to specify this in \texttt{read.table()}).
		\item One row for each individual.
		\item First column contains individual names (use \texttt{row.names=1} to specify this in \texttt{read.table()}).
		\item One column for each locus.
	\end{enumerate}
\end{itemize}
\medskip

%%%%%%%%%%%%%%%%%%%%%%%
\section{Acknowledgements}           %%
%%%%%%%%%%%%%%%%%%%%%%%

The authors would like to thank the Ohio Supercomputer Center for access to computing resources and Nick Skomrock for assistance with deriving the full conditional distributions of the model in the diploid case. We would also like to thank Frederic Austerlitz, Aaron Wenzel, members of the Wolfe and Kubatko labs and 3 anonymous reviewers for their helpful comments on the manuscript. This work was partially funded through a grant from the National Science Foundation (DEB-1455399) to ADW and LSK.
\medskip


%%%%%%%%%%%%%%%%%
% References                       %%
%%%%%%%%%%%%%%%%%

\singlespacing

\bibliographystyle{MolEcol}
\bibliography{refs}

\doublespacing

%%%%%%%%%%%%%%%%%%%%%%%
\section{Author Contributions}        %%
%%%%%%%%%%%%%%%%%%%%%%%

Conceived of the study: PDB, LSK and ADW. PDB derived the polyploid model, ran the simulations and other analyses, coded the R package and wrote the initial draft of the manuscript. PDB, LSK and ADW reviewed all parts of the manuscript and all authors approved of the final version.
\medskip

%%%%%%%%%%%%%%%%%%%%%%
\section{Data Accessibility}            %%
%%%%%%%%%%%%%%%%%%%%%%

Scripts for simulating the data sets, analyzing them using Gibbs sampling and producing the figures from the resulting output can all be found on GitHub, along with the original simulated data sets and autotetraploid potato data (\url{https://github.com/pblischak/polyfreqs-ms-data}). We also provide an implementation of the Gibbs sampler for estimating allele frequencies in the R package \textsc{polyfreqs} (\url{https://github.com/pblischak/polyfreqs}). See the package vignette or GitHub wiki for more details (\url{https://github.com/pblischak/polyfreqs/wiki}).
%\newpage
%\vspace{2in} % Only way we could get Data Accessibility section to fit before Table 1 w/o using a linebreak.

%%%%%%%%%%
% Tables          %%
%%%%%%%%%%

\begin{table}
\centering
\rowcolors{1}{white}{gray!25}
\caption{Notation and symbols used in the description of the model for estimating allele frequencies in polyploids. Vector and matrix forms of the variables are also provided when appropriate.}
\vspace{0.25in}
\bgroup
\def\arraystretch{1.45}
\begin{tabular}[l]{l | l}
\hline
\textbf{Symbol} & \textbf{Description}\\ \hline
$L$ & The number of loci. \\
$\ell$ & Index for loci ($\ell\; \in \{1,\ldots,L\}$). \\
$N$ & Total number of individuals sequenced. \\
$i$ & Index for individuals ($i\; \in \{1,\ldots,N\}$). \\
$\psi$ & The ploidy level of individuals in the population (e.g., tetraploid: $\psi$=4). \\
$p_{\ell}$ & Frequency of the reference allele at locus $\ell$. [$\bm{p}$] \\
$g_{i \ell}$ & The number of copies of the reference allele for individual $i$ at locus $\ell$. [$\bm{G}$] \\
$\tilde{g}_{i \ell}$ & Simulated genotype for posterior predictive model checking. \\
$g_\epsilon$ & The probability of observing a reference read corrected for sequencing error. \\
$t_{i \ell}$ & The total number of reads for individual $i$ at locus $\ell$. [$\bm{T}$] \\
$r_{i \ell}$ & The number of reads with the reference allele for individual $i$ at locus $\ell$. [$\bm{R}$] \\
$\tilde{r}_{i \ell}$ & Simulated reference read count for posterior predictive model checking. \\
$\epsilon$ & Sequencing error. \\
$\mathcal{H}_e, \mathcal{H}_o$ & Expected and observed heterozygosity.\\
\hline
\end{tabular}
\egroup
\label{table1}
\vspace{0.25in}
\end{table}

%%%%%%%%%%%
% Figures           %%
%%%%%%%%%%%

\begin{figure}
\centering
\caption{Error in allele frequency estimation as measured by the RMSE of posterior means. Columns represent the different allele frequencies used to simulate read data (0.01, 0.05, 0.1, 0.2, 0.4), rows represent the number of individuals samples from the population (5, 10, 20, 30). Each individual plot shows the RMSE of the estimates for each ploidy level (tetra, hex, octo) across the different levels of coverage (5x, 10x, 20x, 50x 100x). The best scenario is in the bottom left with 30 individuals sampled and an allele frequency of 0.01. The worst scenario is in the upper right corner with 5 individuals sampled and an allele frequency of 0.4. Looking across rows shows that error increases as allele frequencies get closer to 0.5. Looking up and down columns shows that error increases as the number of individuals decreases. Within each plot, increasing sequence coverage does not have as large of an effect on error, and differences in ploidy show that error decreases as ploidy increases.}
\vspace{0.25in}
\includegraphics{eps/fig1}
\label{fig1:rmse}
\end{figure}

\begin{figure}
\centering
\caption{The posterior standard deviation for allele frequencies decreases compared across levels of sequencing coverage. This plot provides a comparison of the distribution of the posterior standard deviations of the 100 replicates performed for each level of sequencing coverage (5x, 10x, 20x, 50x, 100x) for the hexaploid simulation with 30 individuals sampled from the population and an allele frequency of 0.2.}
\vspace{0.25in}
\includegraphics{eps/fig2}
\label{fig2:coverage-sd}
\end{figure}


\begin{figure}
\centering
\caption{Posterior distributions of the multi-locus estimates of expected and observed heterozygosity in \textit{Solanum tuberosum}. The observed heterozygosity is higher than the expected, consistent with a pattern of excess outbreeding.}
\vspace{0.25in}
\includegraphics{eps/fig3}
\label{fig3:het}
\end{figure}

\end{document}